homeassistant/components/siren/__init__.py

```diff
@@ -63,11 +63,10 @@ def process_turn_on_params(
 
     if not supported_features & SUPPORT_TONES:
         params.pop(ATTR_TONE, None)
-    elif ATTR_TONE in params and (
-        not siren.available_tones
-        or (tone := params[ATTR_TONE]) not in siren.available_tones
+    elif (tone := params.get(ATTR_TONE)) is not None and (
+        not siren.available_tones or tone not in siren.available_tones
```

Comment:
We're still missing a test for the first case on this line, ie no available tones.

Reply:
thanks, lost it in my revert: https://github.com/home-assistant/core/pull/53014